### ISSUE TITLE
Agents are killed after 10 minutes

### ISSUE THREAD

**Issue by maxbalan** [gocd/kubernetes-elastic-agents#193]
**GOCD:** 20.3.0
**kubernetes-elastic-agents-plugin**: 3.7.0-217
Hi, I am following the setup in of k8s elastic agent in documentation, and I noticed that if the job runs more than 10 min the agent is killed even though the job was not finished.
I tried to setup a simple job that prints to console then sleeps for 30sec and the behaviour seems consistent all elastic agents are killed after 10 min mark.

This really affects us from migrating all projects to GOCD as most of them have a lifespan of more than 10 min.

**Comment by varshavaradarajan** [comment]
@maxbalan  - do you have a job timeout configured either in the job settings or globally in the server configuration? https://docs.gocd.org/current/configuration/job_timeout.html#introduction

**Comment by maxbalan** [comment]
hi @varshavaradarajan yes I do have a global timeout and its 60 min  and the all the jobs are using global timeout, I did even try for this test to set the job to never timeout but still the agent is killed after 10 min

**Comment by arvindsv** [comment]
@maxbalan By any chance, do you have two GoCD servers up, pointing to the same k8s cluster?

**Comment by maxbalan** [comment]
@arvindsv no I have only one instance of gocd-server and its sits on its own vm outside k8s

**Comment by arvindsv** [comment]
Hmm. Anything in the plugin logs regarding those jobs? What about the pod's events? What is the reason for it being terminated?

**Comment by maxbalan** [comment]
@arvindsv last time I monitored this issue there was no logs at all in gocd, if you need any specific logs please let me know how to collect them and I will be happy to assist and reproduce the issue

**Comment by maxbalan** [comment]
@arvindsv @varshavaradarajan  after my last comment I tried to reproduce the issue, though now I am running `gocd 20.4.0` as we upgraded it to the latest version, and cannot reproduce this issue any more, I guess it had to do something with v20.3.0, in any case I think we can close this ticket now. I will keep an eye on it in case it start again. I did setup a job for it and will let it run for a couple of days to make sure it was not just a lucky run.

Thank you both for jumping in and trying to help
